### PR TITLE
Fold ClassFromMethod into method records

### PR DESCRIPTION
**Pull request by jdmpapin**
This series changes records that define method symbols to include a defining class ID, so that they can define both the method and its defining class. Previously, (almost) all method records needed a separate `ClassFromMethod` record to tie the method to its class. Including this as part of the method record is much more space-efficient. `ClassFromMethod` is removed because it is no longer necessary

### CODE REVIEW

**Comment by dsouzai** [comment]
@aviansie-ben fyi for the changes in J9AheadOfTimeCompile.cpp

**Comment by dsouzai** [comment]
Jenkins test sanity xlinux,win,plinux,zlinux jdk8,jdk11

**Comment by dsouzai** [comment]
Preemptively kicking off jenkins builds since I'm fairly certain there is a valid answer to my question about the virtual call offset; I'll hold off the approval until then.

File: runtime/compiler/codegen/J9AheadOfTimeCompile.cpp [hunk 1]
```diff
@@ -1545,11 +1548,12 @@ J9::AheadOfTimeCompile::dumpRelocationData()
...
             if (isVerbose)
                {
-               traceMsg(self()->comp(), "\n Validate Virtual Method From Offset: methodID=%d, beholderID=%d, virtualCallOffset=%d, ignoreRtResolve=%s ",
+               traceMsg(self()->comp(), "\n Validate Virtual Method From Offset: methodID=%d, definingClassID=%d, beholderID=%d, virtualCallOffset=%d, ignoreRtResolve=%s ",
                         (uint32_t)binaryTemplate->_methodID,
+                        (uint32_t)binaryTemplate->_definingClassID,
                         (uint32_t)binaryTemplate->_beholderID,
-                        binaryTemplate->_virtualCallOffset,
-                        binaryTemplate->_ignoreRtResolve ? "true" : "false");
+                        binaryTemplate->_virtualCallOffsetAndIgnoreRtResolve & ~1,
```
[the remark this answers is not in the document]
Comment (jdmpapin) [reply runtime/compiler/codegen/J9AheadOfTimeCompile.cpp]: The virtual call offset is the negative offset we add to the address of a receiver's `J9Class` to find the JIT vtable entry, which holds the jitted entry point of that class's implementation. Both the `J9Class` and the JIT vtable entry are aligned to pointer-width boundaries, so the offset is a multiple of the pointer size, which is even. There is a [new `SVM_ASSERT`](https://github.com/eclipse/openj9/pull/4529/files?utf8=%E2%9C%93&diff=unified#diff-3e7f7c3bf56a075860c3736280f74b05R558) in `addVirtualMethodFromOffsetRecord()` to check that the bit is free.

**Comment by dsouzai** [comment]
Jenkins test sanity plinux jdk8

**Comment by dsouzai** [comment]
jdk11 plinux passed all tests, but failed with
```
java.io.EOFException
	at java.io.ObjectInputStream$PeekInputStream.readFully(ObjectInputStream.java:2681)
	at java.io.ObjectInputStream$BlockDataInputStream.readShort(ObjectInputStream.java:3156)
	at java.io.ObjectInputStream.readStreamHeader(ObjectInputStream.java:862)
	at java.io.ObjectInputStream.<init>(ObjectInputStream.java:358)
	at hudson.remoting.ObjectInputStreamEx.<init>(ObjectInputStreamEx.java:49)
	at hudson.remoting.Command.readFrom(Command.java:140)
	at hudson.remoting.Command.readFrom(Command.java:126)
	at hudson.remoting.AbstractSynchronousByteArrayCommandTransport.read(AbstractSynchronousByteArrayCommandTransport.java:36)
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:63)
Caused: java.io.IOException: Unexpected termination of the channel
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:77)
Caused: hudson.remoting.ChannelClosedException: Channel "unknown": Remote call on ub16-ppcle-3 failed. The channel is closing down or has closed down
	at hudson.remoting.Channel.call(Channel.java:948)
	at hudson.FilePath.act(FilePath.java:1072)
	at hudson.FilePath.act(FilePath.java:1061)
	at hudson.FilePath.mkdirs(FilePath.java:1246)
	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:79)
	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:67)
	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution$1$1.call(SynchronousNonBlockingStepExecution.java:50)
	at hudson.security.ACL.impersonate(ACL.java:290)
	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution$1.run(SynchronousNonBlockingStepExecution.java:47)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```
jdk8 failed with the same jenkins error but during the process of javacing the tests, so I've kicked that off in the hope that it passes.

**Pull request merged**